client/src/helpers/APICalls/profileListService.ts

```diff
@@ -0,0 +1,26 @@
+import getMock from '../../mocks/mockProfile';
+import { Profile } from '../../interface/Profile';
+import { DayRange } from './../../interface/Profile';
+
+interface ProfileResponse {
+  profiles: Profile[];
+  uniqueAddress: string[];
+}
+// eslint-disable-next-line
+export async function getList(addressQuery: string | undefined, range: DayRange): Promise<ProfileResponse> {
+  const mockProfiles = getMock().filter((profile) => {
+    if (addressQuery) return profile.address.toLowerCase().includes(addressQuery.toLowerCase());
+    return true;
+  });
+  return await Promise.resolve({
+    profiles: mockProfiles,
+    uniqueAddress: getUniqueAddress(mockProfiles),
+  });
+}
+
+// The returned list of unique addresses is used to populate a combo box for a more directed search
+function getUniqueAddress(profiles: Profile[]): string[] {
+  const addressList = profiles.map((profile) => profile.address);
+  const uniqueAddress = new Set(addressList);
+  return Array.from(uniqueAddress);
+}
```

Comment:
This logic extracts an array of unique addresses from the returned profiles to use as `options` prop in the `Autocomplete` component of the profile Listings page. This makes for a more efficient search experience.

 This logic will be moved to the server when work is done on the integration. However, this can become computationally expensive as the app grows. Perhaps later, a separate collection can be added to store unique addresses in the database.